vars/buildDockerAndPublishImage.groovy

```diff
@@ -0,0 +1,159 @@
+def call(String imageName, Map config=[:]) {
+  if (!config.registry) {
+    if (infra.isTrusted() || infra.isInfra()) {
+      config.registry = "jenkinsciinfra/"
+    } else {
+      config.registry = "jenkins4eval/"
+    }
+  }
+
+  if (!config.dockerfile) {
+    config.dockerfile = "Dockerfile"
+  }
+
+  pipeline {
+    agent {
+      kubernetes {
+        label 'helmfile'
+        inheritFrom 'jnlp-linux'
+        yaml """
```

Comment:
It would be nice to have the YAML file in a dedicated *.yaml file in this repository so we easily automate file update using something similar to [updateCliPodTemplates.tpl](ttps://github.com/jenkins-infra/charts/blob/b79e9e7298c755a34f807626c1d7aa01501e1d55/updateCli/updateCli.d/updateCliPodTemplate.tpl#L42)

Reply:
I wasn't sure how the template would work with the shared library. Does it need to be in each repo or just here?

So played it safe